File: leaf_procurement/leaf_procurement/overrides/driver.py
import uuid
import frappe
from erpnext.setup.doctype.driver.driver import Driver  # type: ignore
from frappe.model.naming import make_autoname


class CustomDriver(Driver):
    def before_insert(doc):
        if not doc.custom_sync_id:
            doc.custom_sync_id = str(uuid.uuid4())    
       
    def autoname(self):
        if getattr(self, "skip_autoname", False):
            print("Skipping autoname for CustomDriver", self.servername)
            self.name = self.servername
            return

        print("Running autoname for CustomDriver", self.name)
        from datetime import datetime

        year = datetime.today().strftime('%Y')

        settings = frappe.get_doc("Leaf Procurement Settings")
        self.custom_location = settings.get("location_warehouse")
        self.custom_code = frappe.db.get_value(
            "Warehouse",
            settings.get("location_warehouse"),
            "custom_short_code"
        )

        prefix = f"{self.custom_code}-DRV-{year}-"
        self.name = make_autoname(prefix + ".#####")

File: leaf_procurement/public/js/driver.js
frappe.ui.form.on('Driver', {
    //custom driver client script
    onload: function (frm) {
        frm.set_df_property('naming_series', 'hidden', 1);
        if (!frm.doc.custom_location_warehouse) {
            frappe.call({
                method: 'frappe.client.get',
                args: {
                    doctype: 'Leaf Procurement Settings',
                    name: 'Leaf Procurement Settings'
                },
                callback: function (r) {
                    if (r.message) {
                        frm.set_value('custom_location', r.message.location_warehouse);
                    }
                }
            });
        }
    },
});